Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,640 @@
+        if (error != NSCalculationNoError) {
+            INVALIDATE_DECIMAL(result);
+            return error;
+        }
+
+        // set the result back
+        _InternalDecimalCopy(result, &internalResult);
+        NSDecimalCompact(result);
+        return NSCalculationNoError;
+    }
+
+    return _handleDifferenceOfSignAddition(result, &left, &right, roundingMode);
+}
+
+/**
+ @Status Caveat
+ @Notes roundingMode is not supported.
+*/
+NSCalculationError NSDecimalSubtract(NSDecimal* result,
+                                     const NSDecimal* leftOperand,
+                                     const NSDecimal* rightOperand,
+                                     NSRoundingMode roundingMode) {
+    THROW_NS_IF_FALSE(E_INVALIDARG, (leftOperand != NULL) && (rightOperand != NULL) && (result != NULL));
+    VALIDATE_ARITHMETIC_INPUT(result, leftOperand, rightOperand);
+    // Convert the subtraction into an addition.
+    // L - R -> L + (-R)
+    // flip the sign, if and only if it's non zero and a valid number
+    NSDecimal right;
+    NSDecimalCopy(&right, rightOperand);
+    right._isNegative = (IS_ZERO(&right) || NSDecimalIsNotANumber(&right)) ? right._isNegative : !right._isNegative;
+
+    return NSDecimalAdd(result, leftOperand, &right, roundingMode);
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiply(NSDecimal* result,
+                                     const NSDecimal* leftOperand,
+                                     const NSDecimal* rightOperand,
+                                     NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalDivide(NSDecimal* result,
+                                   const NSDecimal* leftOperand,
+                                   const NSDecimal* rightOperand,
+                                   NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSString* NSDecimalString(const NSDecimal* dcm, id locale) STUB_METHOD {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+}
+
+static void _InternalDecimalCopy(NSDecimal* destination, NSDecimalInternal* source) {
+    if (!source || !destination) {
+        return;
+    }
+    NSDecimalCopy(destination, &(source->decimal));
```

Comment:
LOL

---
In reply to: [81245870](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81245870) [](ancestors = 81245870)